test/Dialect/FIRRTL/layers.mlir

```diff
@@ -0,0 +1,63 @@
+// RUN: circt-opt %s | FileCheck %s
+
+firrtl.circuit "Test" {
+  firrtl.module @Test() {}
+
+  // CHECK-LABEL: firrtl.layer @A bind
+  firrtl.layer @A bind {}
+
+  // CHECK-LABEL: firrtl.module @WhenUnderLayer(in %test: !firrtl.uint<1>) {
+  // CHECK-NEXT:    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  // CHECK-NEXT:    firrtl.layerblock @A {
+  // CHECK-NEXT:      %w = firrtl.wire : !firrtl.uint<1>
+  // CHECK-NEXT:      firrtl.when %test : !firrtl.uint<1> {
+  // CHECK-NEXT:        firrtl.strictconnect %w, %c0_ui1 : !firrtl.uint<1>
+  // CHECK-NEXT:      }
+  // CHECK-NEXT:    }
+  // CHECK-NEXT:  }
+  firrtl.module @WhenUnderLayer(in %test: !firrtl.uint<1>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.layerblock @A {
+      %w = firrtl.wire : !firrtl.uint<1>
+      firrtl.when %test : !firrtl.uint<1> {
+        firrtl.strictconnect %w, %c0_ui1 : !firrtl.uint<1>
+      }
+    }
+  }
+
+  // CHECK-LABEL: firrtl.module @ProbeEscapeLayer(out %p: !firrtl.probe<uint<1>, @A>) {
+  // CHECK-NEXT:    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  // CHECK-NEXT:    firrtl.layerblock @A {
+  // CHECK-NEXT:      %0 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+  // CHECK-NEXT:      %1 = firrtl.ref.cast %0 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+  // CHECK-NEXT:      firrtl.ref.define %p, %1 : !firrtl.probe<uint<1>, @A>
+  // CHECK-NEXT:    }
+  // CHECK-NEXT:  }
+  firrtl.module @ProbeEscapeLayer(out %p: !firrtl.probe<uint<1>, @A>) {
+    %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+    firrtl.layerblock @A {
+      %0 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+      %1 = firrtl.ref.cast %0 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+      firrtl.ref.define %p, %1 : !firrtl.probe<uint<1>, @A>
+    }
+  }
+
+  // CHECK-LABEL: firrtl.module @Layers4(out %o: !firrtl.openbundle<p: probe<uint<1>, @A>>) {
+  // CHECK-NEXT:    firrtl.layerblock @A {
+  // CHECK-NEXT:      %0 = firrtl.opensubfield %o[p] : !firrtl.openbundle<p: probe<uint<1>, @A>>
+  // CHECK-NEXT:      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  // CHECK-NEXT:      %1 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+  // CHECK-NEXT:      %2 = firrtl.ref.cast %1 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+  // CHECK-NEXT:      firrtl.ref.define %0, %2 : !firrtl.probe<uint<1>, @A>
+  // CHECK-NEXT:    }
+  // CHECK-NEXT:  }
+  firrtl.module @Layers4(out %o: !firrtl.openbundle<p: probe<uint<1>, @A>>) {
+    firrtl.layerblock @A {
+      %0 = firrtl.opensubfield %o[p] : !firrtl.openbundle<p: probe<uint<1>, @A>>
+      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+      %1 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+      %2 = firrtl.ref.cast %1 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+      firrtl.ref.define %0, %2 : !firrtl.probe<uint<1>, @A>
+    }
+  }
+}
```

Comment:
There's nothing to actually test here, unless the PR is trying to test the round-tripping. However, that is/should be handled elsewhere. I'd expect to see this file, as written, not be passed through FileCheck.